backend/state.py

```diff
@@ -413,19 +420,28 @@ def step(self, actions):
             AssertionError: If the action is invalid with the given arguments in
             the given state.
         """
+        self.movement.step(self, clock)
+
         for action, param_arg_dict in actions:
             if not action:
                 continue
             assert action.is_valid(self, param_arg_dict)
-            self = action.perform_action(self, param_arg_dict)
+            # If the action is a movement action, use the movement module
+            if action.name == "move":
```

Comment:
I tried separating the logic for the step functions, so that the movement handles the logic. Currently, movement.step i still called within state.step, but movement.step handles all movement related functions and state.step handles everything else